tensorflow/compiler/xla/python/types.h

```diff
@@ -118,6 +118,7 @@ pybind11::tuple SpanToTuple(absl::Span<int64_t const> xs);
 template <typename T>
 std::vector<T> IterableToVector(const pybind11::iterable& iterable) {
   std::vector<T> output;
+  output.reserve(std::distance(iterable.begin(), iterable.end()));
```

Comment:
I'm afraid that traversing `iterable` to compute the distance is O(N) here?

Reply:
Yeah I was worried about that which is why I didn't send the PR back when I did the whole set of these kinds of PRs. I was trying to figure out if there was some other way to get the size, see my Nov 9 question: https://gitter.im/pybind/Lobby?at=618b285d98c13e7550277c49

Reply:
@joker-eph So do you think this would be reasonably refactored as a `py::sequence`? - Or should I just close this PR…

Reply:
If you convert to pybind11::sequence, and all the tests are passing, this is likely fine.

Reply:
@SamuelMarks Can you please check @joker-eph's comments and keep us posted ? Thanks!